c7n/cli.py

```diff
@@ -39,21 +39,76 @@ def _default_options(p):
                    help="Only execute named/matched policies")
     p.add_argument("-t", "--resource", default=None, dest='resource_type',
                    help="Only execute policies with the given resource type")
+
     p.add_argument("-v", "--verbose", action="store_true",
                    help="Verbose Logging")
+
+    p.add_argument("--debug", action="store_true",
+                   help="Dev Debug")
+
+    if extended_options:
+        _extended_default_options(p)
+
+
+def _extended_default_options(p):
     p.add_argument(
         "-l", "--log-group", default=None,
         help="Cloudwatch Log Group to send policy logs")
 
-    p.add_argument("--debug", action="store_true",
-                   help="Dev Debug")
     p.add_argument("-s", "--output-dir", required=True,
                    help="Directory or S3 URL For Policy Output")
     p.add_argument("-f", "--cache", default="~/.cache/cloud-custodian.cache")
     p.add_argument("--cache-period", default=60, type=int,
                    help="Cache validity in seconds (Default 60)")
 
 
+def _policy_metrics_options(p):
+    """ Add options specific to policy-metrics subcommand. """
+    _default_options(p, extended_options=False)
+    p.add_argument(
+        '--start', type=date_parse, help='Start date (requires --end, overrides --days)')
+    p.add_argument(
+        '--end', type=date_parse, help='End date')
+    p.add_argument(
+        '--days', type=int, default=14,
+        help='Number of days of history to consider (default: %(default)i)')
+    p.add_argument('--period', type=int, default=60*24*24)
+
+    # The original tools/policymetrics.py set these to None, so I am adding them
+    # as hidden options so they are always None.
+    p.add_argument("--log-group", default=None, help=argparse.SUPPRESS)
+    p.add_argument("--cache", default=None, help=argparse.SUPPRESS)
+
+
+def _policy_metrics_validate(parser, options):
+    """ Validate options specified for policy-metrics subcommand. """
+
+    # --start and --end must be specified together
+    if bool(options.start) ^ bool(options.end):
+        # Using `parser.exit` instead of `parser.error` because the latter one
+        # will print the top-level usage message, instead of the policy-metrics
+        # specific one.  I couldn't figure out how to get argparse to give me
+        # the subparser class so I can't call the more appropriate
+        # `subparser.error`.
+        parser.exit(status=2, message='error: --start and --end must be specified together.\n')
+
+
+def _schema_options(p):
+    """ Add options specific to schema subcommand. """
+
+    p.add_argument('resource', metavar='selector', nargs='?', default=None)
+    p.add_argument(
+        '--summary', action="store_true",
+        help="Summarize counts of available resources, actions and filters")
```

Comment:
Do you want it removed from the listed options on the help menu? (i.e. using suppression as shown above)